holoviews/tests/ui/bokeh/test_callback.py

```diff
@@ -495,36 +530,218 @@ def popup_form(name):
     expect(locator).to_be_visible()
 
 
+@skip_popup
+@pytest.mark.parametrize("tool", ["box_select", "lasso_select", "tap"])
+@pytest.mark.parametrize("popup_position", [
+    "top_right", "top_left", "bottom_left", "bottom_right",
+    "right", "left", "top", "bottom"
+])
+@pytest.mark.usefixtures("bokeh_backend")
+def test_stream_popup_position_selection1d(serve_hv, points, tool, popup_position):
+    def popup_form(index):
+        if index:
+            return f"# selection\n{len(index)} {popup_position}"
+
+    hv.streams.Selection1D(source=points, popup=popup_form, popup_position=popup_position)
+    points.opts(tools=[tool], active_tools=[tool])
+
+    page = serve_hv(points)
+    hv_plot = page.locator('.bk-events')
+    expect(hv_plot).to_have_count(1)
+
+    box = hv_plot.bounding_box()
+    if tool == "box_select":
```

Comment:
Maybe we could extract this if statement into a helper function (in this file), something like `def _select_points_based_on_tool(tool=..., page=page, plot=hv_plot)`.